data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretsSupplier.java

```diff
@@ -107,17 +108,26 @@ public void refresh(String secretConfigId) {
         LOG.info("Finished retrieving latest secret in aws:secrets:{}.", secretConfigId);
     }
 
+    @SkipTestCoverageGenerated
     private Object retrieveSecretsFromSecretManager(final AwsSecretManagerConfiguration awsSecretManagerConfiguration,
                                                     final SecretsManagerClient secretsManagerClient) {
         final GetSecretValueRequest getSecretValueRequest = awsSecretManagerConfiguration
                 .createGetSecretValueRequest();
-        final GetSecretValueResponse getSecretValueResponse;
-        try {
-            getSecretValueResponse = secretsManagerClient.getSecretValue(getSecretValueRequest);
-        } catch (Exception e) {
-            throw new RuntimeException(
-                    String.format("Unable to retrieve secret: %s",
-                            awsSecretManagerConfiguration.getAwsSecretId()), e);
+        GetSecretValueResponse getSecretValueResponse;
+        while (true) {
```

Comment:
I don't think having an infinite loop this deep is a good idea. We should manage this in a thread specifically for this purpose.